src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```diff
@@ -125,10 +134,23 @@ public ParseStatus parseJavadocAsDetailNode(DetailAST javadocCommentAst) {
             result.setTree(tree);
         }
         catch (ParseCancellationException | IllegalArgumentException ex) {
-            // If syntax error occurs then message is printed by error listener
-            // and parser throws this runtime exception to stop parsing.
-            // Just stop processing current Javadoc comment.
-            ParseErrorMessage parseErrorMessage = errorListener.getErrorMessage();
+            ParseErrorMessage parseErrorMessage = null;
+
+            if (ex.getCause() instanceof FailedPredicateException
+                    || ex.getCause() instanceof NoViableAltException) {
```

Comment:
This input [InputWrongHtmlTagOrder.javadoc](https://github.com/checkstyle/checkstyle/pull/4643/files#diff-cf650261cbc45957536374934264a426) would result in a `FailedPredicateException`. Because of entirely removing the alternative which contained `notifyErrorListener(...)`, cases like `<tag1> xyz </tag2>` were getting parsed, so I had to use predicate `{isSameTagNames($htmlElementOpen.ctx, $htmlElementClose.ctx)}?` in there. The test which check messages for `no viable alternative` have nothing to do with this branch.